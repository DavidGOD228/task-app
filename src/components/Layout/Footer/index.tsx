import { Box, Text } from '@chakra-ui/react';
import React from 'react';

export default function Footer() {
  return (
    <Box
      as='footer'
      width='100%'
      position='fixed'
      bottom='0'
      left='0'
      p={4}
      bgColor='gray.100'
      shadow='sm'
      zIndex='docked'
    >
      <Text>Â© 2024 Test App</Text>
    </Box>
  );
}
