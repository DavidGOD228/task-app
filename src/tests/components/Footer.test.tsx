import React from 'react';
import { render, screen } from '@testing-library/react';

import '@testing-library/jest-dom';
import Footer from '@/components/Layout/Footer';

describe('Footer Component', () => {
  test('renders footer with the correct text', () => {
    render(<Footer />);
    expect(screen.getByText('Â© 2024 Test App')).toBeInTheDocument();
  });

  test('footer snapshot', () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });
});
