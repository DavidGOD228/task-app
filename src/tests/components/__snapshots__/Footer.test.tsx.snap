// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component footer snapshot 1`] = `
<DocumentFragment>
  <footer
    class="css-mr2up3"
  >
    <p
      class="chakra-text css-0"
    >
      Â© 2024 Test App
    </p>
  </footer>
</DocumentFragment>
`;
